src/metax_api/models/catalog_record.py

```diff
@@ -1649,6 +1773,40 @@ def _get_preferred_identifier_type_from_request(self):
             pid_type = IdentifierType.URN
         return pid_type
 
+    def change_cumulative_state(self, new_value):
+        if self.next_dataset_version:
+            raise Http400('Cannot change old dataset version')
+
+        if self.cumulative_state is not self.CUMULATIVE_STATE_NO and new_value == self.CUMULATIVE_STATE_NO:
+            raise Http400('Cumulative dataset cannot be set to non-cumulative dataset. '
+                        'If you want to stop active cumulation, set cumulative status to closed.')
+
+        if self.cumulative_state == self.CUMULATIVE_STATE_NO and new_value == self.CUMULATIVE_STATE_CLOSED:
+            raise Http400('Cumulation cannot be closed for non-cumulative dataset')
+
+        if new_value == self.CUMULATIVE_STATE_YES and self.preservation_state > self.PRESERVATION_STATE_INITIALIZED:
+            raise Http400('Cumulative datasets are not allowed in PAS process. Change preservation_state '
+                        'to 0 in order to change the dataset to cumulative.')
+
+        self.date_modified = get_tz_aware_now_without_micros()
+
+        if new_value == self.CUMULATIVE_STATE_CLOSED:
+            self.date_cumulation_ended = self.date_modified
+
+        if self.cumulative_state != self.CUMULATIVE_STATE_YES and new_value == self.CUMULATIVE_STATE_YES:
+            # create new version
+            file_changes = self._find_file_changes()
+            self._create_temp_record(file_changes, cumulative_change=True)
+            self._create_new_dataset_version(new_value)
```

Comment:
Move cumulative -related stuff from method ``_create_new_dataset_version`` to be handled here before calling it, so that ``_create_new_dataset_version`` does not really know anything about cumulative stuff. If I am not mistaken, it should be enough that state is set to yes, plus cumulation start date.